app/src/main/java/com/github/se/eventradar/ui/viewProfile/ProfileUi.kt

```diff
@@ -113,68 +161,183 @@ fun ProfileUi(
+                Column(modifier = Modifier.fillMaxWidth().testTag("leftAlignedViewProfileColumn")) {
+                  OutlinedTextField(
+                      value = uiState.bio,
+                      onValueChange = viewModel::onBioChanged,
+                      modifier =
+                          Modifier.fillMaxWidth().padding(top = 10.dp).testTag("bioTextField"),
+                      label = { Text("Bio") },
+                      isError = uiState.bioIsError,
+                      colors =
+                          OutlinedTextFieldDefaults.colors()
+                              .copy(
+                                  focusedPrefixColor = MaterialTheme.colorScheme.primary,
+                                  unfocusedLabelColor = MaterialTheme.colorScheme.primary,
+                              ))
+                  Row(
+                      horizontalArrangement = Arrangement.SpaceBetween,
+                      modifier = Modifier.padding(top = 10.dp).testTag("phoneNumberBirthDateRow")) {
+                        OutlinedTextField(
+                            value = uiState.phoneNumber,
+                            onValueChange = viewModel::onPhoneNumberChanged,
+                            modifier = Modifier.weight(1f).testTag("phoneNumberTextField"),
+                            label = { Text("Phone Number") },
+                            isError = uiState.phoneNumberIsError,
+                            colors =
+                                OutlinedTextFieldDefaults.colors()
+                                    .copy(
+                                        focusedPrefixColor = MaterialTheme.colorScheme.primary,
+                                        unfocusedLabelColor = MaterialTheme.colorScheme.primary,
+                                    ))
+                        Spacer(
+                            modifier = Modifier.width(10.dp).testTag("phoneNumberBirthDateSpacer"))
+                        OutlinedTextField(
+                            value = uiState.birthDate,
+                            onValueChange = viewModel::onBirthDateChanged,
+                            modifier = Modifier.weight(1f).testTag("birthDateTextField"),
+                            label = { Text("Birth Date") },
+                            isError = uiState.birthDateIsError,
+                            colors =
+                                OutlinedTextFieldDefaults.colors()
+                                    .copy(
+                                        focusedPrefixColor = MaterialTheme.colorScheme.primary,
+                                        unfocusedLabelColor = MaterialTheme.colorScheme.primary,
+                                    ))
+                      }
+                }
+
+                Button(
+                    onClick = {
+                      if (viewModel.validateFields()) {
+                        viewModel.updateUserInfo()
+                        isEditMode.value = false
```

Comment:
Currently, when the user clicks the goBack button (which I saw sets the isEditMode.value to false), it somehow triggers this? and the user information still gets saved, maybe a fix could be adding a state in the VM for isCancelSelected (which is updated to true when user clicks go back button). I'm not sure if this would fix it all though but hope it helps!

Reply:
I implemented the goBackButton to save the changes only in the UI state if the user presses the back function. If he decides to finish editing his profile, he can continue where he left off and if he leaves the changes will be lost. I added a Toast message saying the User Info was not saved. I hope this helps clear up the behavior.

Reply:
This looks great. Only thing is the unsaved changes get removed when I switch tabs - is there a way to keep the unsaved changes across the entire app lifecycle, unless the user logs out?

Reply:
Noted that this is too difficult to change in the limited time we have. The behaviour will have to remain like this.